File: src/domain/claude_domain.py
from domain.abstracts.base_domain import BaseDomain

class Claude(BaseDomain):

    def __init__(self, server, model_name):        
        super().__init__(server=server, model_name=model_name)
        self.model = "claude-haiku-4-5-20251001"

    def build_response_messages(self, response) -> str:
        return "\n".join([r.text for r in response.content])
        
    def send_message(self, prompt: str) -> str:
        return self.send(lambda: self.server.messages.create(
                model=self.model,
                messages=[{"role": "user", "content": prompt}],
                max_tokens=self.max_tokens,
            ))

    def list_models(self):
        try:
            return self._get_domain_view("=== Claude Models ===")
        except Exception as e:
            return "=== Claude Models ===", [f"[ERROR] {e}"], ""

    def _fetch_domain_names(self) -> list[str]:
        models = self.server.models.list()
        return [m.id for m in models.data]

File: angular-app/src/core/domain/claude-domain.ts

import { IServer, ModelDescriptor, TextCompletionResponse } from '../infrastructure/servers';
import { BaseDomain } from './abstracts/base-domain';

export class Claude extends BaseDomain {
  override readonly model = 'claude-3-5-haiku-20241022';

  constructor(server: IServer, modelName: string) {
    super(server, modelName);
  }

  override buildResponseMessages(response: unknown): string {
    const completion = response as TextCompletionResponse;
    if (Array.isArray(completion.content)) {
      return completion.content.map((chunk) => chunk.text || chunk.content || '').join('\n');
    }
    return String(completion.content ?? '');
  }

  override async sendMessage(prompt: string): Promise<string> {
    return this.send(() => this.server.messages!.create({
      model: this.model,
      messages: [{ role: 'user', content: prompt }],
      max_tokens: this.maxTokens,
    }));
  }

  override async listModels(): Promise<string[]> {
    try {
      return await this.getDomainNamesCached();
    } catch (e) {
      return [`[ERROR] ${String(e)}`];
    }
  }

  protected async _fetchDomainNames(): Promise<string[]> {
    const models = await this.server.models!.list();
    return models.data.map((model: ModelDescriptor) => model.id);
  }
}
